code/game/objects/effects/mines.dm

```diff
@@ -116,8 +118,6 @@
 	to_chat(victim, "<span class='danger'>*click*</span>")
 
 /obj/effect/mine/proc/on_entered(datum/source, atom/movable/AM)
-	SIGNAL_HANDLER
```

Comment:
You can't just remove signal handler from a signal handler. Use addtimer instead of sleep.

Reply:
fixed, and as an added bonus I also found and fixed a bug with smartmine detection.